Add notes about native unions. (#33698)
dotnet/docs: docs/standard/native-interop/customize-struct-marshalling.md

};
```

### Unions

> A union is a data type that can contain different types of data atop the same memory. It is a common form of data in C. A union can be expressed in .NET using `LayoutKind.Explicit`. It recommended to use structs when defining a union in .NET. Using classes can cause issues with layout and produce unpredictable behavior.

Reviewer: ```suggestion
A union is a data type that can contain different types of data atop the same memory. It's a common form of data in the C language. A union can be expressed in .NET using `LayoutKind.Explicit`. It's recommended to use structs when defining a union in .NET. Using classes can cause layout issues and produce unpredictable behavior.
```
